sdk-cpp/include/do_error_macros.h

```diff
@@ -4,49 +4,30 @@
 #ifndef _DELIVERY_OPTIMIZATION_DO_ERROR_MACROS_H
 #define _DELIVERY_OPTIMIZATION_DO_ERROR_MACROS_H
 
-/*
-TODO(jimson): Move these error macros into a shared header for this project
-These error macros are defined behind guards so that they do not redefine the macro for for platforms/toolchains in which they already exist
-When moving into a shared file, definitions for some macros in doclient are behind the docli namespace, and they also log errors - sdk has no notion of logging, so logging macros should be no-ops 
-*/
+#include "do_errors.h"
 
 #include <stdint.h>
 
 namespace microsoft
 {
 namespace deliveryoptimization
 {
-
-#ifndef HRESULT
-using HRESULT = int32_t;
-#endif
-
-#ifndef S_OK
-#define S_OK            ((HRESULT)0L)
-#endif
-
-#ifndef RETURN_IF_FAILED
-#define RETURN_IF_FAILED(hr)  {  \
-    int32_t __hr = (hr);            \
-    if(FAILED(__hr)) return __hr;   \
-}
-#endif
-
 #ifndef FACILITY_DELIVERY_OPTIMIZATION 
 #define FACILITY_DELIVERY_OPTIMIZATION   208
 #endif
 
-//TODO(jimson) Looks like these conversions are causing test issues with tests
+//TODO(jimson): Do we really need these after switching to error code class? The error_category should help us differentiate what type of error it is
```

Comment:
>need [](http://example.com/codeflow?start=29&length=4)

Agree we don't need these anymore.

Reply:
ah nvm, thinking about your comment in do_errors.h regarding using error_category

Reply:
Let's still apply the facility code where needed for DU purposes, and make use of error_category